### PR TITLE
docs(agentApp): Examples on how to add a history view for your human agent

### PR DESCRIPTION
**Pull request by ethanwinters**
Requires web chat 4.5.0 to be published first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,10 @@ The `serviceDeskFactory` configuration setting expects a factory that returns an
 
 One of the items passed into the factory is a callback object. These callbacks are defined in [./src/types/serviceDeskCallback.ts](./src/types/serviceDeskCallback.ts). These are the functions you will call inside your service desk code to communicate information back to the web chat.
 
+### Displaying the chat history to your human agent
+
+As of the 4.5.0 release of web chat, Watson Assistant will pass configuration needed to display a chat history widget in you live agent application interface. This agent application will contain a copy of the conversation your customer had with Watson Assistant for your live agent to be able to view. Visit ['./docs/AGENT_APP.md'](./docs/AGENT_APP.md) to learn more.
```
Comment (rrberry) [inline README.md:107]:
```suggestion
As of the 4.5.0 release of web chat, Watson Assistant will pass configuration needed to display a chat history widget in your live agent application interface. This agent application will contain a copy of the conversation your customer had with Watson Assistant for your live agent to be able to view. Visit ['./docs/AGENT_APP.md'](./docs/AGENT_APP.md) to learn more.
```

File: docs/AGENT_APP.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Embedded chat history widget
+
+The embedded agent application allows your live agents to view a chat history of the end users conversation with Watson Assistant to better understand their needs.
```
Comment (rrberry) [inline docs/AGENT_APP.md:3]:
```suggestion
The embedded agent application allows your live agents to view a chat history of the end user's conversation with Watson Assistant to better understand their needs.
```

File: docs/AGENT_APP.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## How it works
+
+Starting with web chat 4.5.0, as part of the [startChat](./API.md#startChat) method a second parameter is passed to the startChat method. This `startChatOptions` parameter is an object with options and metadata about the chat. See [the StartChatOptions type](../src/types/serviceDesk.ts) for more information.
```
Comment (rrberry) [inline docs/AGENT_APP.md:7]:
```suggestion
Starting with web chat 4.5.0, as part of the [startChat](./API.md#startChat) method, a second parameter is passed to the startChat method. This `startChatOptions` parameter is an object with options and metadata about the chat. See [the StartChatOptions type](../src/types/serviceDesk.ts) for more information.
```

File: docs/AGENT_APP.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Located at `startChatOptions.agentAppInfo.sessionHistoryKey` is a string that contains everything web chat needs to securely open a read-only mode of the web chat with a transcript of the end user and Watson Assistant visible. The vast majority of service desks have a way to pass meta information to the agents view and either embed an iFrame or embed HTML and JavaScript. It is up to you to determine how to pass the `sessionHistoryKey` to either the IBM provided iFrame or JavaScript.
```
Comment (rrberry) [inline docs/AGENT_APP.md:9]:
```suggestion
Located at `startChatOptions.agentAppInfo.sessionHistoryKey` is a string that contains everything web chat needs to securely open a read-only mode of the web chat showing a transcript of the conversation between the end user and Watson Assistant. The vast majority of service desks have a way to pass meta information to the agent's view and either embed an iFrame or embed HTML and JavaScript. It is up to you to determine how to pass the `sessionHistoryKey` to either the IBM provided iFrame or JavaScript.
```

File: docs/AGENT_APP.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## Viewing agent application via iFrame
+
+Some service desks allow you to add an iFrame to the agent page via a server side template language. If that mechanism also allows you to access the `sessionHistoryKey` you can simply embed an iFrame that will render the history of the conversation:
```
Comment (rrberry) [inline docs/AGENT_APP.md:13]:
```suggestion
Some service desks allow you to add an iFrame to the agent page via a server side template language. If that mechanism also allows you to access the `sessionHistoryKey`, you can simply embed an iFrame that will render the history of the conversation:
```

File: docs/AGENT_APP.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+```
+<div id="agent-app-host" style="width: 380px; height: 380px;"></div>
+<script src="https://web-chat.global.assistant.watson.appdomain.cloud/loadAgentAppFrame.js" />
+<script>
+  loadAgentApp(sessionHistoryKey, document.querySelector('#agent-app-host'));
+</script>
+```
+
+The `loadAgentApp` method takes two arguments. First, the `sessionHistoryKey`. The second argument is a DOM element for the read-only web chat to render to. The web chat will grow to the size of the element and the element should be given a height and width for the web chat to grow to.
```
Comment (rrberry) [inline docs/AGENT_APP.md:36]:
```suggestion
The `loadAgentApp` method takes two arguments. The first argument is the `sessionHistoryKey` value. The second is a DOM element for the read-only web chat to render to. The web chat will grow to the size of the element and the element should be given a height and width for the web chat to grow to.
```

File: docs/API.md [hunk 7]
```diff
@@ -7,14 +7,15 @@ To be implemented by the developer. Implementation template can be found in [../
 ### startChat
 
 ```ts
-async startChat(connectMessage: MessageResponse): Promise<void>
+async startChat(connectMessage: MessageResponse, startChatOptions: StartChatOptions): Promise<void>
 ```
 
-Opens the chat with the service desk and performs any necessary setup steps.
+Opens the chat with the service desk and performs any necessary setup steps. Most service desks have a way to embed a custom iFrame into the agent view, as well as a way to pass metadata into that iFrame. startChatOptions.agentAppInfo contains metadata for you to be able to render the conversation history with Watson Assistant to your agents in a custom iFrame and this data should be passed via whatever methods the service desk you are using uses. See [./AGENT_APP.md](./AGENT_APP.md).
```
Comment (rrberry) [inline docs/API.md:13]:
```suggestion
Opens the chat with the service desk and performs any necessary setup steps. Most service desks have a way to embed a custom iFrame into the agent view, as well as a way to pass metadata into that iFrame. `startChatOptions.agentAppInfo` contains metadata for you to be able to render the conversation history with Watson Assistant to your agents in a custom iFrame, and this data should be passed using whatever methods are appropriate for the service desk you are using. See [./AGENT_APP.md](./AGENT_APP.md).
```

**Review by rrberry — APPROVED** [review]

File: src/types/serviceDesk.ts [hunk 8]
```diff
@@ -121,4 +123,23 @@ interface ServiceDeskStateFromWAC {
...
+interface StartChatOptions {
+  agentAppInfo: AgentAppInfo;
+}
+
+/**
+ * This data is used to configure an read only web chat inside the service desk.
+ */
+interface AgentAppInfo {
+  /**
+   * A string that is separated by that can be used to create a PublicConfig. It includes base connect
```
Comment (TazmanianDI) [inline src/types/serviceDesk.ts:138]: You're missing `":"` here.

**Review by TazmanianDI — APPROVED** [review]

**Review by TazmanianDI — APPROVED** [review]

**Pull request merged**